tests: samples/drivers/uart/async_api: NXP Lpuart platforms build failure

### Describe the bug

still failure on some NXP platforms:

- rddrone_fmuk66/mk66f18
- frdm_k32l2b3/k32l2b31a
- imx95_evk/mimx9596/m7


3c8def069121930a0f82df10de0d20e0545a3e81 is the first bad commit
```
commit 3c8def069121930a0f82df10de0d20e0545a3e81
Author: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>
Date:   Thu Aug 7 15:36:48 2025 +0200

    samples: drivers: uart: async_api: add new scenario for console harness
    
    - Add a console harness to filter console output and validate tests.
    
    Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

 samples/drivers/uart/async_api/sample.yaml | 14 ++++++++++++++
 1 file changed, 14 insertions(+)
``


### Regression

- [x] This is a regression.

### Steps to reproduce

```
scripts/twister -p frdm_mcxc242 -T samples/drivers/uart/async_api --build-only
```

### Relevant log output

```shell

<!--StartFragment-->


[101/161]   Linking C static library zephyr/drivers/dma/libdrivers__dma.a[102/161]   Building C object   zephyr/drivers/gpio/CMakeFiles/drivers__gpio.dir/gpio_mcux.c.obj[103/161]   Linking C static library zephyr/drivers/gpio/libdrivers__gpio.a[104/161]   Building C object   zephyr/drivers/pinctrl/CMakeFiles/drivers__pinctrl.dir/common.c.obj[105/161]   Building C object   zephyr/drivers/pinctrl/CMakeFiles/drivers__pinctrl.dir/pinctrl_nxp_port.c.obj[106/161]   Linking C static library zephyr/drivers/pinctrl/libdrivers__pinctrl.a[107/161]   Building C object   zephyr/drivers/regulator/CMakeFiles/drivers__regulator.dir/regulator_common.c.obj[108/161]   Building C object   zephyr/drivers/regulator/CMakeFiles/drivers__regulator.dir/regulator_fixed.c.obj[109/161]   Linking C static library   zephyr/drivers/regulator/libdrivers__regulator.a[110/161] Building C object   zephyr/drivers/serial/CMakeFiles/drivers__serial.dir/uart_mcux.c.obj[111/161]   Building C object   zephyr/drivers/serial/CMakeFiles/drivers__serial.dir/uart_mcux_lpuart.c.objFAILED:   zephyr/drivers/serial/CMakeFiles/drivers__serial.dir/uart_mcux_lpuart.c.obj y   -Wno-format-zero-length -Wdouble-promotion -Wno-pointer-sign -Wpointer-arith   -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int   -fno-pic -fno-pie -fno-asynchronous-unwind-tables -ftls-model=local-exec   -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop   -fmacro-prefix-map=/home/jenkins/agent/workspace/zephyr_downstream_build_daily/zephyr/samples/drivers/uart/async_api=CMAKE_SOURCE_DIR   -fmacro-prefix-map=/home/jenkins/agent/workspace/zephyr_downstream_build_daily/zephyr=ZEPHYR_BASE   -fmacro-prefix-map=/home/jenkins/agent/workspace/zephyr_downstream_build_daily=WEST_TOPDIR   -ffunction-sections -fdata-sections -specs=picolibc.specs -std=c99 -MD -MT   zephyr/drivers/serial/CMakeFiles/drivers__serial.dir/uart_mcux_lpuart.c.obj   -MF   zephyr/drivers/serial/CMakeFiles/drivers__serial.dir/uart_mcux_lpuart.c.obj.d   -o   zephyr/drivers/serial/CMakeFiles/drivers__serial.dir/uart_mcux_lpuart.c.obj   -c /home/jenkins/agent/workspace/zephyr_downstream_build_daily/zephyr/drivers/serial/uart_mcux_lpuart.c/home/jenkins/agent/workspace/zephyr_downstream_build_daily/zephyr/drivers/serial/uart_mcux_lpuart.c:   In function   'mcux_lpuart_configure_async':/home/jenkins/agent/workspace/zephyr_downstream_build_daily/zephyr/drivers/serial/uart_mcux_lpuart.c:1134:16:   error: implicit declaration of function 'LPUART_GetRxFifoCount'   [-Werror=implicit-function-declaration] 1134 \| while   (LPUART_GetRxFifoCount(config->base) > 0) { \| ^~~~~~~~~~~~~~~~~~~~~cc1:   all warnings being treated as errorsninja: build stopped: subcommand failed.
--
```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

- OS: ubuntu
- tool-chain: 0.17.1
- version: 65d7c7c598a3486455d234e404e07f9055aac780

### Additional Context

_No response_

--- a/drivers/serial/Kconfig.mcux_lpuart
+++ b/drivers/serial/Kconfig.mcux_lpuart
@@ -1,6 +1,4 @@
-# MCUXpresso SDK LPUART
-
-# Copyright (c) 2017, NXP
+# Copyright 2017, 2025 NXP
 # SPDX-License-Identifier: Apache-2.0
 
 config UART_MCUX_LPUART
@@ -10,7 +8,7 @@ config UART_MCUX_LPUART
 	depends on CLOCK_CONTROL
 	select SERIAL_HAS_DRIVER
 	select SERIAL_SUPPORT_INTERRUPT
-	select SERIAL_SUPPORT_ASYNC
+	select SERIAL_SUPPORT_ASYNC if $(dt_compat_any_has_prop,$(DT_COMPAT_NXP_LPUART),dmas)
 	select DMA if UART_ASYNC_API
 	select PINCTRL
 	help

